tests/integration/cloud/helpers/cloud_test_base.py

```diff
@@ -60,73 +73,63 @@ def assertInstanceExists(self, creation_ret=None, instance_name=None):
-                break
-            else:
-                sleep(self.__RE_RUN_DELAY)
-                query = self.query_instances()
-
-        # Assert that the last query was successful
-        self.assertTrue(self._instance_exists(instance_name, query),
-                        'Instance "{}" was not created successfully: |\n\t{}\n\t|`'.format(
-                            instance_name, '\n\t'.join(creation_ret if creation_ret else query)))
-
-        log.debug('Instance exists and was created: "{}"'.format(instance_name))
-
         # If it exists but doesn't show up in the creation_ret, there was probably an error during creation
         if creation_ret:
             self.assertIn(instance_name, [i.strip(': ') for i in creation_ret],
                           'An error occured during instance creation:  |\n\t{}\n\t|'.format(
-                              '\n\t'.join(creation_ret if creation_ret else query)
+                              '\n\t'.join(creation_ret)
                           ))
+        else:
+            # Verify that the instance exists via query
+            query = self.query_instances()
+            for tries in range(self.__RE_TRIES):
+                if self._instance_exists(instance_name, query):
+                    log.debug(
+                        'Instance "{}" reported after {} seconds'.format(instance_name, tries * self.__RE_RUN_DELAY))
+                    break
+                else:
+                    sleep(self.__RE_RUN_DELAY)
+                    query = self.query_instances()
+
+            # Assert that the last query was successful
+            self.assertTrue(self._instance_exists(instance_name, query),
+                            'Instance "{}" was not created successfully: '.format(', '.join(query)))
+
+            log.debug('Instance exists and was created: "{}"'.format(instance_name))
 
     def _destroy_instance(self):
+        shutdown_delay = 30
         log.debug('Deleting instance "{}"'.format(self.instance_name))
         delete = self.run_cloud('-d {0} --assume-yes'.format(self.instance_name), timeout=TIMEOUT)
         # example response: ['gce-config:', '----------', '    gce:', '----------', 'cloud-test-dq4e6c:', 'True', '']
-        delete_str = ''.join(delete)
-        log.debug('Deletion status: {}'.format(delete_str))
+        delete_str = yaml.safe_dump(delete).strip()
+        log.debug('Deletion status: |\n{}'.format(delete_str))
 
         if any([x in delete_str for x in (
             'True',
             'was successfully deleted'
         )]):
+            destroyed = True
             log.debug('Instance "{}" was successfully deleted'.format(self.instance_name))
         elif any([x in delete_str for x in (
             'shutting-down',
             '.delete',
+            self.instance_name + '-DEL'
         )]):
+            destroyed = True
             log.debug('Instance "{}" is cleaning up'.format(self.instance_name))
```

Comment:
It will probably also complain about this line too, even though it isn't touched.